functional/util/util.go

```diff
@@ -154,3 +154,32 @@ func WaitForState(stateCheckFunc func() bool) (time.Duration, error) {
 		}
 	}
 }
+
+// SliceContains returns true if every string in needle slice exists in any
+// slice in haystack.
+func SliceContains(haystack []string, needle []string) bool {
+	for _, n := range needle {
+		foundNeedle := false
+		for _, h := range haystack {
+			if strings.Compare(n, h) == 0 {
+				foundNeedle = true
+				break
+			}
+		}
+		if !foundNeedle {
+			return false
+		}
+	}
+
+	return true
+}
+
+// GetListUnitNames returns the 1st field of the given string, which is
+// a single line of output of "fleetctl list-{units,unit-files}".
+func GetListUnitNames(listUnitFiles []string) []string {
```

Comment:
That's a good point! so since it's just functional maybe return a map unless some other points that we are not aware of...